### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by tjmehta**

### CODE REVIEW

File: lib/create-host-label.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+require('bind-right'); // fn.bindRight
+var pluck = require('101/pluck');
+var add = function (a, b) {
```
Comment (cflynn07) [inline lib/create-host-label.js]: nit, thought our convention was not to assign functions to variables

File: lib/runnable-hostname.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  validateOpts(opts);
+  // consistency btw direct and elastic
+  return direct(opts).replace(opts.shortHash+'-', '');
+}
+
+function direct (opts) {
+  validateOpts(opts);
+  var template = '{shortHash}-{repoName}-{env}-{ownerUsername}'
+    .replace('{shortHash}', opts.shortHash)
+    .replace('{env}', 'staging');
```
Comment (cflynn07) [inline lib/runnable-hostname.js:21]: hardcode staging?

Comment (cflynn07) [inline lib/runnable-hostname.js:21]: I know urls are case-insensitive but do we want to tack on a toLowerCase() just for the heck of it?

File: lib/runnable-hostname.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      key: 'ownerUsername',
+      val: opts.ownerUsername,
+      max: 39 // maxlength of username in github is 39
+    },
+    {
+      key: 'repoName',
+      val: getRepoName(opts)
+    }
+  ]);
+  return (subdomain+'.'+opts.userContentDomain);
```
Comment (cflynn07) [inline lib/runnable-hostname.js:33]: toLowerCase here? ^

File: lib/runnable-hostname.js (hunk 2 above)
Comment (tjmehta) [inline lib/runnable-hostname.js:21]: I didn't toLowerCase it bc I want to frontend to maintain how it currently displays urls with the proper casing

**Comment by cflynn07** [comment]
just noticing some drift in terminology, serverName (runnable-api-client) vs instanceName (here)
https://github.com/CodeNow/runnable-api-client/commit/8de6d5e4a0f33c6de488f4d21940ec0d309067b0#diff-46e2f4d4632851d7517273192c28d272R221

**Comment by tjmehta** [comment]
It should be `instanceName`

**Pull request merged**